### PR TITLE
Add support for carthage

### LINKED ISSUE #137: Carthage Support?
I'll take a look into setting up the project with a PR, but wanted to leave the issue here.

### PR DESCRIPTION
**Pull request by pthariensflame**
Closes #137, at least for now.  This is necessary for my work on [Aquifer](http://github.com/pthariensflame/Aquifer).

### CODE REVIEW

**Comment by pthariensflame** [comment]
Don't merge this yet.  I'm still working on how to make Carthage behave.

**Comment by pthariensflame** [comment]
This is now ready!  The only thing is that Carthage can't directly build Swiftz for some reason, so any users have to attach it as a dependent subproject to their own.  It'll still fetch it, place it in its proper location, and keep it up to date, though.  :)

File: swiftz.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -633,7 +630,7 @@
 			isa = PBXProject;
 			attributes = {
 				LastUpgradeCheck = 0610;
-				ORGANIZATIONNAME = "Maxwell Swadling";
+				ORGANIZATIONNAME = TypeLift;
```
Comment (CodaFi) [inline swiftz.xcodeproj/project.pbxproj]: :+1:

Comment (pthariensflame) [inline swiftz.xcodeproj/project.pbxproj]: :)

**Comment by CodaFi** [comment]
The same thing is happening with the Basis.  Something about the testing bundle (I suspect codesigning) is causing xcodebuild to crap out.  Weird that both cases work with Travis but not that

File: swiftz_core/swiftz_core/Error.swift [hunk 2]
```diff
@@ -7,8 +7,10 @@
 //
 
 /// Immediately terminates the program with an error message.
+///
+/// This currently only works in Debug mode, since the underlying assertion is optimized away in Release mode.
```
Comment (CodaFi) [inline swiftz_core/swiftz_core/Error.swift]: Does Swift let us use __builtin_unreachable()?  That won't get optimized away in release.

**Comment by CodaFi** [comment]
Before we do this, if you want to rename the target we had best do it here and now.  Perhaps capital Swiftz?

**Comment by pthariensflame** [comment]
Sure! I'll do this as soon as I'm back at my keyboard.  :)

**Comment by CodaFi** [comment]
Seems only one of the targets got a name change.  No matter.  We can sort this out later when I get around to part 2 or #116.

**Comment by CodaFi** [comment]
That is to say :boat:

**Pull request merged**